ragna/deploy/_ui/central_view.py

```diff
@@ -182,24 +182,75 @@ def __init__(self, api_wrapper, **params):
         )
         self.on_click_chat_info = None
 
+    def convert_metadata_filters_to_human(self, metadata_filters):
```

Comment:
This is for the chat info side bar. Since we don't have proper design for this, can we maybe just use the default `__repr__` of the metadata filter?

https://github.com/Quansight/ragna/blob/aea6b6706265e342e41682a88ea80bface4961ec/ragna/core/_metadata_filter.py#L31-L46

It's not too bad.

Reply:
By using this : 
```metadata_filters_readable = MetadataFilter.from_primitive(self.current_chat["metadata"]["input"])```

I get the following result : 
![Capture d’écran 2024-08-14 à 09 06 48](https://github.com/user-attachments/assets/dfd39f74-564e-42a0-b225-02414e425c1f)

For comparison, the same filters in the current state : 
![Capture d’écran 2024-08-14 à 09 07 22](https://github.com/user-attachments/assets/ae0c87bc-6483-45da-9ddc-0144af114eda)

What do you think is best ?

Reply:
In the first one, new lines are not displayed:

```python
from ragna.core import MetadataFilter

print(
    MetadataFilter.and_(
        [
            MetadataFilter.eq("document_name", "test"),
            MetadataFilter.gt("size", 100),
        ]
    )
)
```

```
AND(
  EQ('document_name', 'test'),
  GT('size', 100),
)
```

This is already an improvement. I think your solution suffers from the same problem: if you have more than two filters, it will be hard to display them in one line.

But that is somewhat besides the point: instead of writing a lot of logic that we'll have to maintain to create another human readable format, we could just go with what is already there. I know it is sometimes hard to let go of code that you have already written, but the benefit currently does not outweigh its maintenance cost.

Nothing is lost though. If we ultimately decide to go with your solution, we have the implementation in the commits.

Reply:
I have accepted Philip's suggestion in commit `47a199`.